src/test/validation_chainstate_tests.cpp

```diff
@@ -72,4 +75,66 @@ BOOST_AUTO_TEST_CASE(validation_chainstate_resize_caches)
     WITH_LOCK(::cs_main, manager.Unload());
 }
 
+//! Test UpdateTip behavior for both active and background chainstates.
+//!
+//! When run on the background chainstate, UpdateTip should do a subset
+//! of what it does for the active chainstate.
+BOOST_FIXTURE_TEST_CASE(chainstate_update_tip, TestChain100Setup)
+{
+    ChainstateManager& chainman = *Assert(m_node.chainman);
+    uint256 curr_tip = ::g_best_block;
+
+    // Mine 10 more blocks, putting at us height 110 where a valid assumeutxo value can
+    // be found.
+    mineBlocks(10);
+
+    // After adding some blocks to the tip, best block should have changed.
+    BOOST_CHECK(::g_best_block != curr_tip);
+
+    BOOST_REQUIRE(CreateAndActivateUTXOSnapshot(m_node, m_path_root));
+
+    // Ensure our active chain is the snapshot chainstate.
+    BOOST_CHECK(chainman.IsSnapshotActive());
+
+    curr_tip = ::g_best_block;
+
+    // Mine a new block on top of the activated snapshot chainstate.
+    mineBlocks(1);  // Defined in TestChain100Setup.
+
+    // After adding some blocks to the snapshot tip, best block should have changed.
+    BOOST_CHECK(::g_best_block != curr_tip);
+
+    curr_tip = ::g_best_block;
+
+    CChainState& background_cs = m_node.chainman->ValidatedChainstate();
+    BOOST_CHECK(m_node.chainman->IsBackgroundIBD(&background_cs));
+
+    // Create a block to append to the validation chain.
+    std::vector<CMutableTransaction> noTxns;
+    CScript scriptPubKey = CScript() << ToByteVector(coinbaseKey.GetPubKey()) << OP_CHECKSIG;
+    CBlock validation_block = this->CreateBlock(noTxns, scriptPubKey, &background_cs);
+    auto pblock = std::make_shared<const CBlock>(validation_block);
+    BlockValidationState state;
+    CBlockIndex* pindex = nullptr;
+    const CChainParams& chainparams = Params();
+    bool newblock = false;
+
+    // TODO: much of this is inlined from ProcessNewBlock(); just reuse PNB()
+    // once it is changed to support multiple chainstates.
+    {
+        LOCK(::cs_main);
+        bool checked = CheckBlock(*pblock, state, chainparams.GetConsensus());
+        BOOST_CHECK(checked);
+        bool accepted = background_cs.AcceptBlock(
+            pblock, state, chainparams, &pindex, true, nullptr, &newblock);
+        BOOST_CHECK(accepted);
+    }
+    bool block_added = background_cs.ActivateBestChain(state, chainparams, pblock);
```

Comment:
In commit "test: validation: add unittest for UpdateTip behavior" (3f1346a0178afef20db3599cc545dbc54200d635)

Since this test is designed to check `UpdateTip` background behavior, but the test doesn't call `UpdateTip` directly, it may be good to add a comment saying this line is what calls `UpdateTip`.

Also could maybe a simple check could be added here to verify the new background tip hash?

Also curious if there's an existing test (or a planned one) to cover the background chain catching up to the activated snapshot.

Reply:
Done. 

> Also curious if there's an existing test (or a planned one) to cover the background chain catching up to the activated snapshot.

Yes; I'll look at writing a test for this sometime today in addition to some chainstate-concurrency torture tests.